### ISSUE TITLE
Mechanics can Repair Rides That are Closed

### ISSUE THREAD

**Issue by Derpercoaster** [OpenRCT2/OpenRCT2#18739]
### Operating System

WIndows 10

### OpenRCT2 build

v0.4.2 (8ceea45)

### Base game

RollerCoaster Tycoon 2

### Describe the issue

Mechanics can repair closed rides, even if the ride has never been tested or opened. This does not occur in the base game, meaning it has to be a bug. It also feels incredibly unfair, as the reliability can be increased back to 100% by leaving a ride closed, and letting mechanics inspect it repeatedly.

### Steps to reproduce

1. Build or have a ride in the park. I used a classic wooden coaster, but it seems to work with all other rides as well.
2. Connect the exit to a path, so mechanics can access the ride.
3. Wait for a mechanic to inspect it. Any inspection time works.
4. This should not happen. This does not happen in RCT2 either. I do not know if it occurs in RCTC.

**Comment by kurbus** [comment]
This could just be intended behavior to make the game easier.

**Comment by ocalhoun6** [comment]
You mean mechanics *inspecting* a ride that's never been opened or tested, right?

Yeah, I have seen it a few times, and it's a bit odd to see a mechanic inspecting the ride when I'm not even finished building it yet.

**Comment by kurbus** [comment]
> You mean mechanics _inspecting_ a ride that's never been opened or tested, right?
> 
> Yeah, I have seen it a few times, and it's a bit odd to see a mechanic inspecting the ride when I'm not even finished building it yet.

well that sucks, I only implemented them not being able to fix the rides on the pretense that's what they were talking about.

**Comment by Gymnasiast** [comment]
This whole report makes very little sense to me.

> Mechanics can repair closed rides, even if the ride has never been tested or opened

How? For that, a ride would have to be broken, and rides cannot break down when they’re closed.

> It also feels incredibly unfair, as the reliability can be increased back to 100% by leaving a ride closed, and letting mechanics inspect it repeatedly.

How is that unfair? Of course something that isn’t used but inspected regularly would be in good shape.

**Comment by ocalhoun6** [comment]
> > You mean mechanics _inspecting_ a ride that's never been opened or tested, right?
> > Yeah, I have seen it a few times, and it's a bit odd to see a mechanic inspecting the ride when I'm not even finished building it yet.
> 
> well that sucks, I only implemented them not being able to fix the rides on the pretense that's what they were talking about. edit: fixed

Yeah... A ride that's never been opened or tested can't break down, so if a mechanic is heading to that ride to *fix* it, then something has gone very, very wrong.

I *have* noticed mechanics *inspecting* rides that were never opened or tested, though. Including several times while I was still actively building the ride. That's a bit of an oddity, but not really a huge deal and doesn't affect much in the game. From a realism perspective, it even sort of makes sense that park mechanics would be somewhat involved in inspecting the ride as it's being built. And they would certainly inspect the ride after it was finished, but before opening it to the public.

Hm... In fact, you know, this should be *very* easy to replicate...
Yep!
![Barony Bridge 2022-12-01 17-30-49](https://user-images.githubusercontent.com/26573303/205194362-b1e7f499-0962-4439-851b-ac04b2b8d6c8.png)
I built one of my own saved prebuilts: "splatter" and never opened it, never tested it, never even tested it with ghost trains. All I did was connect the exit path. Then I hired 1 mechanic and just waited while fast-forwarding.
And ... yep. Eventually, the mechanic headed to "splatter" for an inspection.

**Comment by Derpercoaster** [comment]
I meant to say "inspect". Totally my fault for being tired.

**Comment by Derpercoaster** [comment]
> I think the heart of the issue is probably how the ride is counting 'time since last inspection'. That time should probably _not_ increase if the ride has never been opened or tested, but it seems to be doing so ... and therefore, the never-opened ride eventually calls for an inspection from a mechanic.

This would make sense to me. I think it would be a fun easter egg for mechanics to inspect unfinished or never-opened rides, just for no reason, but they would not inspect closed rides that have been run at least once. It would be fun for a new player to discover an easter egg like that.
